### ISSUE TITLE
question regarding multi clients unlearning

### ISSUE THREAD

**Issue by nikhoj** [Liar-Mask/FedAU#2]
`weight_ul=(1-alpha)*ul_state_dict['classifier.weight']
bias_ul=(1-alpha)*ul_state_dict['classifier.bias']
for idx in self.ul_clients:'
       'weight_ul= alpha / int(len(self.ul_clients)) * ul_state_dicts[idx]['classifier_ul.weight']
        bias_ul= alpha / int(len(self.ul_clients)) * ul_state_dicts[idx]['classifier_ul.bias']`
        
   for multi clients unlearning in your main_zgx is this part of your code correct? the for loop is only considering the last clients weight, correct me if I'm wrong. and is that alpha switched with (1-alpha)?

**Comment by Liar-Mask** [comment]
Thanks for your question. The for loop part you see is the unformal single client unlearning module test content before releasing the code which was not carefully checked out due to my negligence. Sorry. In fact, the for loop part should be:
`weight_ul += alpha / int(len(self.ul_clients)) * ul_state_dicts[idx]['classifier_ul.weight']`
`bias_ul += alpha / int(len(self.ul_clients)) * ul_state_dicts[idx]['classifier_ul.bias']`

Besides, the `alpha` here is actually 1-alpha in the paper, but as it does not affect the use and is easier to control the role of the unlearning module, we will reserve this usage in the code.
I've updated the loop part and fixed other running issues. Thanks for the heads up.
